pkg/products/threescale/reconciler.go

```diff
@@ -1327,37 +1356,19 @@ func (r *Reconciler) reconcileOutgoingEmailAddress(ctx context.Context, serverCl
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
-	var existingSMTPFromAddress string
-	if r.installation.Status.CustomSmtp != nil && r.installation.Status.CustomSmtp.Enabled {
-		existingSMTPFromAddress, err = cs.GetFromAddress(ctx, serverClient, r.installation.Namespace)
-
-		if err != nil {
-			r.log.Error("error getting smtp_from address from custom smtp secret", err)
-			return integreatlyv1alpha1.PhaseFailed, err
-		}
-	} else {
-		existingSMTPFromAddress, err = resources.GetExistingSMTPFromAddress(ctx, serverClient, observabilityConfig.GetNamespace())
-
-		if err != nil {
-			if !k8serr.IsNotFound(err) {
-				r.log.Error("Error getting smtp_from address from secret alertmanager-application-monitoring", err)
-				return integreatlyv1alpha1.PhaseFailed, nil
-			}
-			r.log.Warning("failure finding secret alertmanager-application-monitoring: " + err.Error())
-		}
-	}
-	if existingSMTPFromAddress == "" {
-		r.log.Warning("Couldn't find SMTP in a secret, retrieving it from the envar")
-		existingSMTPFromAddress = os.Getenv(integreatlyv1alpha1.EnvKeyAlertSMTPFrom)
+	existingSMTPFromAddress, err := resources.GetSMTPFromAddress(ctx, serverClient, r.log, r.installation, observabilityConfig.GetNamespace())
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
 	}
 	accessToken, err := r.GetAdminToken(ctx, serverClient)
 	if err != nil {
 		r.log.Info("Failed to get admin token in reconcileOutgoingEmailAddresss: " + err.Error())
 		return integreatlyv1alpha1.PhaseInProgress, err
 	}
 
-	if integreatlyv1alpha1.IsRHOAMMultitenant(integreatlyv1alpha1.InstallationType(r.installation.Spec.Type)) {
-		existingSMTPFromAddress = "test@rhmw.io"
+	err = r.reconcileTenantOutgoingEmailAddress(ctx, serverClient, existingSMTPFromAddress)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
 	_, err = r.tsClient.SetFromEmailAddress(existingSMTPFromAddress, *accessToken)
```

Comment:
Might be out of scope, but would you know if setting the `from_email` field in the smtp secret from https://github.com/3scale/3scale-operator/blob/master/doc/apimanager-reference.md#system-smtp, set the `from` address value for the no-reply mail and in the tenants ? If so, do we want this ? 🤔

Reply:
A follow up ticket is good with me, was just wondering if you knew it 👍